dbcon/execplan/windowfunctioncolumn.cpp

```diff
@@ -661,18 +670,35 @@ void WindowFunctionColumn::evaluate(Row& row, bool& isNull)
                     break;
                 }
 
-                default:
+                case 8:
                 {
                     if (row.equals<8>(BIGINTNULL, fInputIndex))
                         isNull = true;
                     else
                     {
-                        fResult.decimalVal.value = (int64_t)row.getUintField<8>(fInputIndex);
+                        fResult.decimalVal.value = row.getIntField<8>(fInputIndex);
+                        fResult.decimalVal.scale = (unsigned)fResultType.scale;
```

Comment:
No. output precision is not specified. We don't want to limit the output precision of aggregated values. It could lead to all sorts of unintended truncation. The default precision is 0. This is the way it has worked for many years.